SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -8,11 +8,30 @@
 LOG_FILE=/tmp/makenv.log # File to keep a backlog of this script output
 GIT_HOME="$HOME/SWAN_projects" # Path where git repositories are stored
 
+# Check if an environment already exists in the session, avoiding multiple environments
+CURRENT_ENV_NAME=$(find "/home/$USER" -type d -name "*_env" | head -n 1 | cut -d '/' -f4)
+CURRENT_REPO_PATH=$(tail -n 1 "/home/$USER/.bash_profile" | cut -d ' ' -f2)
+
+# Check if an environment already exists in the session
+if [ -n "${CURRENT_ENV_NAME}" ]; then
+    echo "ENVIRONMENT_ALREADY_EXISTS:${CURRENT_ENV_NAME}"
```

Comment:
Is this new check only necessary now with conda? If it's not related, please in its own commit with its message.

Reply:
This check isn't new. Is was just moved to the begining of the file to avoid running code unnecessarily. This is made for checking if there's any custom environment created already, no matter the name or type.

Reply:
Ok so it's not conda related. It's own commit and message then, please.